src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDataTable.java

```diff
@@ -285,7 +282,7 @@ private void internalAddrow(JDBCType jdbcType, Object val, List<Object> rowValue
                         currentColumnMetadata.precision = nValueLen;
                         columnMetadata.put(pair.getKey(), currentColumnMetadata);
                     }
-                    rowValues.set(key, val);
+                    rowValues[key] = val;
                     break;
 
```

Comment:
At Line [301](https://github.com/microsoft/mssql-jdbc/pull/1121/files#diff-7b7ee3303fcc9546cf8cffdf5e9cecdfR301), in the else block before the condition, you need to assign `rowValues[key] = null`

Reply:
I'm not sure that needed to happen.  rowValues[key] would have already been null since nothing else would have set it.